test/message-service.test.js

```diff
@@ -0,0 +1,47 @@
+describe('Test message service', () => {
+  let messageService
+  let rheaPromiseMock
+
+  beforeEach(async () => {
+    messageService = require('../server/services/message-service')
+    rheaPromiseMock = require('rhea-promise')
+    jest.mock('rhea-promise')
+  })
+
+  test('Message service sends the claim to two queues', async () => {
+    const claimRecord = {
+      claimId: 'MINE123',
+      propertyType: 'business',
+      accessible: false,
+      dateOfSubsidence: new Date(),
+      mineType: ['gold', 'iron']
+    }
+    const jsonData = JSON.stringify(claimRecord)
+    rheaPromiseMock._SendFunction.mockReset()
+    rheaPromiseMock._SendFunction.mockImplementation(function () {
+      return new rheaPromiseMock.Delivery()
+    })
+    await messageService.publishClaim(claimRecord)
+    await expect(rheaPromiseMock._SendFunction).toHaveBeenCalledTimes(2)
+    await expect(rheaPromiseMock._SendFunction).toHaveBeenCalledWith({ body: jsonData })
+  })
+
+  test('Message service acts correctly to an error while sending', async () => {
+    const claimRecord = {
+      claimId: 'MINE123',
+      propertyType: 'business',
+      accessible: false,
+      dateOfSubsidence: new Date(),
+      mineType: ['gold', 'iron']
+    }
+    rheaPromiseMock._SendFunction.mockReset()
+    rheaPromiseMock._SendFunction.mockImplementation(function () {
+      throw new Error()
+    })
+    return expect(messageService.publishClaim(claimRecord)).rejects.toThrow()

```

Comment:
We should test that the correct error is thrown by passing either the error instance (if it should simply be re-thrown by `messageService`) or a string/regex to match the error message (if `messageService` creates and throws its own error object)

Reply:
I've created a ticket to improve testing of errors thrown and functions called with expected arguments: https://eaflood.atlassian.net/browse/FPD-375